projects/cm_mcu/InitTask.c

```diff
@@ -55,8 +56,25 @@ void InitTask(void *parameters)
   // otherwise, block its operations indefinitely until it's available
   acquireI2CSemaphoreBlock(i2c2_sem);
 
+  char progchip_temp[5][CLOCK_PROGNAME_REG_NAME];
+  char progeeprom_temp[5][CLOCK_EEPROM_PROGNAME_REG_NAME];
   for (int i = 0; i < 5; ++i) {
     init_load_clk(i); // load each clock config from EEPROM
+    // get and print out the file name
+    vTaskDelay(pdMS_TO_TICKS(500));
+    char progname_clkdesgid[CLOCK_PROGNAME_REG_NAME];     // program name from DESIGN_ID register of clock chip
+    char progname_eeprom[CLOCK_EEPROM_PROGNAME_REG_NAME]; // program name from eeprom
+    getClockProgram(i, progname_clkdesgid, progname_eeprom);
+    strcpy(progchip_temp[i], progname_clkdesgid);
+    strcpy(progeeprom_temp[i], progname_eeprom);
```

Comment:
use `strncpy` instead to avoid buffer overruns

Reply:
thank you. Done.